beacon-chain/core/altair/sync_committee.go

```diff
@@ -0,0 +1,181 @@
+package altair
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/mathutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+const maxRandomByte = uint64(1<<8 - 1)
+
+// ValidateNilSyncContribution validates the following fields are not nil:
+// -the contribution and proof itself
+// -the message within contribution and proof
+// -the contribution within contribution and proof
+// -the aggregation bits within contribution
+func ValidateNilSyncContribution(s *ethpb.SignedContributionAndProof) error {
+	if s == nil {
+		return errors.New("signed message can't be nil")
+	}
+	if s.Message == nil {
+		return errors.New("signed contribution's message can't be nil")
+	}
+	if s.Message.Contribution == nil {
+		return errors.New("inner contribution can't be nil")
+	}
+	if s.Message.Contribution.AggregationBits == nil {
+		return errors.New("contribution's bitfield can't be nil")
+	}
+	return nil
+}
+
+// NextSyncCommittee returns the next sync committee for a given state.
+//
+// Spec code:
+// def get_next_sync_committee(state: BeaconState) -> SyncCommittee:
+//    """
+//    Return the next sync committee, with possible pubkey duplicates.
+//    """
+//    indices = get_next_sync_committee_indices(state)
+//    pubkeys = [state.validators[index].pubkey for index in indices]
+//    aggregate_pubkey = bls.AggregatePKs(pubkeys)
+//    return SyncCommittee(pubkeys=pubkeys, aggregate_pubkey=aggregate_pubkey)
+func NextSyncCommittee(ctx context.Context, s state.BeaconStateAltair) (*ethpb.SyncCommittee, error) {
+	indices, err := NextSyncCommitteeIndices(ctx, s)
+	if err != nil {
+		return nil, err
+	}
+	pubkeys := make([][]byte, len(indices))
```

Comment:
With regards to my other comment, do you need to validator that len(indices) does not exceed the max committee size?

Reply:
Im leaning towards no. `NextSyncCommittee` is just a converting function on top of `NextSyncCommitteeIndices`. It converts: 

`[]uint{0,1,2,3}` to `SyncCommittee{pubkeys:[]byte{a,b,c,d...`

It shouldn't have knowledge of the committee size, it just converts what `NextSyncCommitteeIndices` returns. `NextSyncCommitteeIndices` could use some other values tomorrow (`SyncCommitteeSizeHardFork100`) and `NextSyncCommittee` wouldn't care